engine/gamesys/src/gamesys/components/comp_mesh.cpp

```diff
@@ -227,6 +256,18 @@ namespace dmGameSystem
         return component->m_Textures[index] ? component->m_Textures[index] : resource->m_Textures[index];
     }
 
+    static inline uint32_t GetVertexSize(const MeshComponent* component) {
+        return component->m_VertexDeclaration ? component->m_VertSize : component->m_Resource->m_VertSize;
+    }
+
+    static inline uint32_t GetElementCount(const MeshComponent* component) {
+        return component->m_BufferResource ? component->m_BufferResource->m_ElementCount : component->m_Resource->m_ElementCount;
+    }
+
+    static inline dmGraphics::HVertexDeclaration GetVertexDeclaration(const MeshComponent* component) {
+        return component->m_VertexDeclaration ? component->m_VertexDeclaration : component->m_Resource->m_VertexDeclaration;
+    }
```

Comment:
The call sites become cleaner when using these helper functions.

Reply:
Indeed. I like these small helper functions. It really does make code more readable!